meshroom/ui/components/filepath.py

```diff
@@ -49,6 +49,12 @@ def extension(self, path):
         """ Returns the extension (.ext) of a pathname """
         return os.path.splitext(self.asStr(path))[-1]
 
+    @Slot(str, result=str)
+    @Slot(QUrl, result=str)
+    def removeExtension(self, path):
+        """ Returns the pathname without its extension (.ext)"""
+        return os.path.splitext(path)[0]
```

Comment:
You should use `self.asStr(path)` as in the methods above to properly handle both strings and QUrls